MapboxDirections/MBDirectionsRouter.swift

```diff
@@ -48,83 +49,65 @@ internal enum MBDirectionsRouter: Router {
         case Full = "full"
     }
     
-    case V4(Configuration, String, [MBDirectionsWaypoint], Bool?, InstructionFormat?, GeometryFormatV4?, Bool?)
-    case V5(Configuration, String, [MBDirectionsWaypoint], Bool?, GeometryFormatV5?, OverviewGranularity?, Bool?, Bool?)
-    
-    var method: HTTPMethod {
-        return .GET
-    }
-    
-    var encoding: HTTPEncoding {
-        return .URL
-    }
-    
-    var configuration: Configuration {
-        switch self {
-        case .V4(let config, _, _, _, _, _, _): return config
-        case .V5(let config, _, _, _, _, _, _, _): return config
-        }
-    }
+    case V4(MBDirectionsConfiguration, String, [MBDirectionsWaypoint], Bool?, InstructionFormat?, GeometryFormatV4?, Bool?)
+    case V5(MBDirectionsConfiguration, String, [MBDirectionsWaypoint], Bool?, GeometryFormatV5?, OverviewGranularity?, Bool?, Bool?)
```

Comment:
I'm not wedded to the original approach of structuring the router around a v4/v5 enum and switch statement. The lack of named parameters was very tricky as I originally wrote this, because the types are totally divorced from the names, much as they might be in idiomatic C code.